src/CALab/GridPanel.java

```diff
@@ -2,9 +2,31 @@
 
 import mvc.AppFactory;
 import mvc.AppPanel;
+import stopLight.StoplightFactory;
+import stopLight.StoplightPanel;
+
+import javax.swing.*;
 
 public class GridPanel extends AppPanel {
-    public GridPanel(GridFactory factory) {
+    private JButton run1;
+    private JButton run50;
+    private JButton populate;
+    private JButton clear;
+    public GridPanel(AppFactory factory) {
```

Comment:
```suggestion
    public GridPanel(GridFactory factory) {
```

Reply:
typing should be based on this class; even though `GridFactory` is an `AppFactory`, we don't want any `AppFactory` getting passed into this constructor.